src/util/logger/file_log.rs

```diff
@@ -75,23 +68,21 @@ impl RotatingFileLogger {
         })
     }
 
-    /// Opens log file with append mode. Creates a new file if it doesn't exist.
-    fn open(&mut self) -> io::Result<()> {
-        self.file = Some(open_log_file(&self.file_path)?);
-        Ok(())
-    }
-
     fn should_rotate(&mut self) -> bool {
         Utc::now() > self.next_rotation_time
     }
 
     /// Rotates the current file and updates the next rotation time.
     fn rotate(&mut self) -> io::Result<()> {
-        self.close()?;
+        self.flush()?;
+
+        // Note: renaming files while they're open only works on Linux and macOS.
```

Comment:
Why not close it? Seems it's more compatible.

Reply:
So that even if the new file failed to be opened for whatever reason we could still write logs to the old file. Plus `File::close` doesn't exist 🙃